### ISSUE TITLE
 Why does three.js appear this warning?

### ISSUE THREAD

**Issue by AlanLLLLL** [vasturiano/3d-force-graph#454]
**Describe the bug**
![image](https://user-images.githubusercontent.com/38875407/113095923-f1ea2a80-9226-11eb-95a4-02d71161fba4.png)

I am not sure if I changed the original data of graphData or the cloneDeep of lodash used？

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

**Comment by vasturiano** [comment]
@AlanLLLLL where are you seeing those warnings? Could you make a quick example on [codepen.io](https://codepen.io/)

**Comment by AlanLLLLL** [comment]
![image](https://user-images.githubusercontent.com/38875407/113232018-f66a1e00-92ce-11eb-9288-a288283e4471.png)
![image](https://user-images.githubusercontent.com/38875407/113232187-3af5b980-92cf-11eb-9eba-aa6601a6b8c5.png)

[https://codepen.io/alanlllll/pen/GRrrKYd?editors=0001](url)
Hi, you can click on the link to have a look. It seems that __threeObj in node does not allow cloneDeep.This warning affects performance very much， I really want to know what caused it, thank you.

**Comment by vasturiano** [comment]
@AlanLLLLL yes your are right. The issue is in running lodash's `cloneDeep` on a Three Mesh. It relies on `length` properties which have a different syntax for Three objects.
You can try with previous versions of `three` to see if it's a recent change or not. 
In any case, probably the proper way to clone a Three object is to invoke its [clone](https://threejs.org/docs/#api/en/objects/Mesh.clone) method.

So if you do have a real need for cloning you whole data structure, you may want to either exclude the `__threeObj` attributes, or handle them differently.

**Comment by AlanLLLLL** [comment]
@vasturiano I tried to delete **__threeobj** and execute **cloneDeep**, but it affects the entire graph, can you give me some suggestions, thanks.

**Comment by vasturiano** [comment]
I meant to exclude the attribute from the clone operation, not mutate the original object. Removing the attribute will certainly cause the graph to misfunction.

Out of curiosity, what's you use case that requires cloning the entire data structure?

**Comment by AlanLLLLL** [comment]
@vasturiano I need to save the position in the scene, add undo and redo operations, so I cloned every piece of data, do you have any better suggestions, thanks.

**Comment by vasturiano** [comment]
Cloning the data structure won't work, because the links point to the node objects by reference, via the `source` and `target` attributes. So simply cloning the data will cause the links to point at the wrong node objects, leading to a disconnected graph.
You can read more about it on the [d3-force-3d docs](https://github.com/vasturiano/d3-force-3d#link_links).

If you'd like to store frames of the nodes positions I would suggest to simply extract the `x`, `y` and `z` attributes of each of the nodes. That should be sufficient to memorize the layout of the graph.

**Comment by AlanLLLLL** [comment]
@vasturiano Ok i will try it, thank you for reply.
